tesla_powerwall/powerwall.py

```diff
@@ -161,6 +162,9 @@ def get_solars(self) -> List[Solar]:
     def get_vin(self) -> str:
         return assert_attribute(self._api.get_config(), "vin", "config")
 
+    def set_island_mode(self, mode: IslandMode) -> IslandMode:
+        return assert_attribute(self._api.post_islanding_mode({"island_mode": mode}), "island_mode")
```

Comment:
```suggestion
        return IslandMode(assert_attribute(self._api.post_islanding_mode({"island_mode": mode.value}), "island_mode"))
```
As `IslandMode` is an Enum it cannot be serialized to json directly, which would result in an Error when POSTing. Instead you must access the `value` which returns the underlying string e.g. "backup" for `IslandMode.ONGRID`. 
Also the value returned must be explicitly converted to an `IslandMode`.

Reply:
Ahh I see, thanks for explaining it! :)